pandas/core/groupby/generic.py

```diff
@@ -1026,38 +1026,64 @@ def _cython_agg_blocks(
         if numeric_only:
             data = data.get_numeric_data(copy=False)
 
-        no_result = object()
-
         def cast_agg_result(result, values: ArrayLike, how: str) -> ArrayLike:
             # see if we can cast the values to the desired dtype
             # this may not be the original dtype
             assert not isinstance(result, DataFrame)
-            assert result is not no_result
 
             dtype = maybe_cast_result_dtype(values.dtype, how)
             result = maybe_downcast_numeric(result, dtype)
 
-            if isinstance(values, ExtensionArray) and isinstance(result, np.ndarray):
-                # e.g. values was an IntegerArray
-                # (1, N) case can occur if values was Categorical
-                #  and result is ndarray[object]
-                # TODO(EA2D): special casing not needed with 2D EAs
-                assert result.ndim == 1 or result.shape[0] == 1
-                try:
-                    # Cast back if feasible
-                    result = type(values)._from_sequence(
-                        result.ravel(), dtype=values.dtype
-                    )
```

Comment:
Why is this no longer needed?

Reply:
because its only relevant for Categorical, and never raises

Reply:
So the comment that values could be IntegerArray wasn't correct anymore? 
And why is `cast_agg_result` no longer be called with other extension arrays?

Reply:
IntegerArray is now handled correctly within _cython_operation

Reply:
OK, I see, thanks for the explanation

But so that means that for a generic EA (not implemented by ourselves), we won't try to cast back? (not sure if there are good use cases for that, but still)

Reply:
> But so that means that for a generic EA (not implemented by ourselves), we won't try to cast back?

Actually, looking at the other PR (https://github.com/pandas-dev/pandas/pull/38162), we simply don't try to apply the cython groupby ops anymore?

Reply:
> we won't try to cast back? (not sure if there are good use cases for that, but still)

i am trying to get rid of that altogether, and only do cast-back in cases where we know it is correct.  If/when _from_sequence is reliably strict we can reconsider.

> Actually, looking at the other PR (#38162), we simply don't try to apply the cython groupby ops anymore?

for an arbitrary EA, no.

Reply:
I don't understand how we can avoid that in general. 

> for an arbitrary EA, no.

That's potentially a regression? What's the reason for this? (now, I don't know how much of the groupby algos support object dtype, or would actually be faster than the python versions if they would)

Reply:
> I don't understand how we can avoid that in general.

I'm not clear what this is referring to.

> That's potentially a regression? What's the reason for this? (now, I don't know how much of the groupby algos support object dtype, or would actually be faster than the python versions if they would)

We fall back to the SeriesGroupBy implementation so there shouldn't be a regression.

Many of these are ordinal-only, so using something like values_for_argsort is likely to be better than casting to object.

Reply:
I was referring to your "i am trying to get rid of that altogether"

Reply:
> I was referring to your "i am trying to get rid of that altogether"

Thanks.

For the functions that go through _cython_operation, there are only a handful of them and we can identify which are dtype-preserving and which are not.  Some custom logic is in `dtypes.cast.maybe_cast_result_dtype`, and the rest at this point is in `_ea_wrap_cython_operation`.  Basically anything that is "custom logic" represents a way we are treating pandas-internal EAs different from 3rd-party, which is something we want to avoid.  Identifying and isolating that custom logic will make it easier to de-custom.

The only other place where the "try to cast back" that i am looking to get rid of altogether is used is in `_aggregate_series_pure_python`, and #38254 removes that (buggy) usage.

Reply:
I don't know if we want no custom handing of our own EAs at all? 
For example what you wrote now in `_ea_wrap_cython_operation`, are this things you want to further remove?

Reply:
> I don't know if we want no custom handing of our own EAs at all?

I mean, in the sense that we also want world peace and unlimited ice cream, yes.  For the foreseeable future we're going to have some non-zero amount of custom handling.

> For example what you wrote now in _ea_wrap_cython_operation, are this things you want to further remove?

In the short run I'm adding handling for FloatingArray and improving the handling for IntegerArray, so not removing at all.